### PR TITLE
Fix srb init issues with ruby-3

### LINKED ISSUE #5332: `srb init` has failed in Ruby 3.1.1
#### Input
Run `srb init` on an [empty repository](https://github.com/mihyaeru21/sorbet-init-error), I got the following errors.

env
- Sorbet 0.5.9663
- Ruby 3.1.1
- Ubuntu 20.04.3

##### Error 1

```
$ bundle exec srb init
...
Would you like to continue? [Y/n] y
Generating: sorbet/config
Generating: sorbet/rbi/sorbet-typed/
Generating: sorbet/rbi/gems/
LoadError: cannot load such file -- sorbet
LoadError: cannot load such file -- sorbet-static
/home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/gem-generator-tracepoint/tracepoint_serializer.rb:35:in `initialize': wrong number of arguments (given 1, expected 0; required keywords: files, delegate_classes) (ArgumentError)
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/gem-generator-tracepoint/tracer.rb:40:in `new'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/gem-generator-tracepoint/tracer.rb:40:in `new'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/gem-generator-tracepoint.rb:43:in `main'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/bin/srb-rbi:232:in `block in make_step'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/bin/srb-rbi:118:in `init'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/bin/srb-rbi:196:in `main'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/bin/srb-rbi:237:in `<main>'
```

This is reported by #5204.
In Ruby 3, `Hash` is not automatically converted to keyword arguments.

Make the following changes to `vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/gem-generator-tracepoint.rb` and move on next error.

```diff
- TracepointSerializer.new(trace_results).serialize(output_dir)
+ TracepointSerializer.new(**trace_results).serialize(output_dir)
```

##### Error 2

```
$ bundle exec srb init
...
Would you like to continue? [Y/n] y
Generating: sorbet/config
Reusing existing config file: sorbet/config
Generating: sorbet/rbi/sorbet-typed/
Generating: sorbet/rbi/gems/
LoadError: cannot load such file -- sorbet
LoadError: cannot load such file -- sorbet-static
Generating: sorbet/rbi/hidden-definitions/
Requiring all of your code
Naming all Modules
Naming Object/home/mihyaeru/.anyenv/envs/rbenv/versions/3.1.1/lib/ruby/3.1.0/set/sorted_set.rb:4:in `rescue in <top (required)>': The `SortedSet` class has been extracted from the `set` library. You must use the `sorted_set` gem or other alternatives. (RuntimeError)
        from /home/mihyaeru/.anyenv/envs/rbenv/versions/3.1.1/lib/ruby/3.1.0/set/sorted_set.rb:1:in `<top (required)>'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/real_stdlib.rb:77:in `require'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/real_stdlib.rb:77:in `const_get'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/real_stdlib.rb:77:in `call'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/real_stdlib.rb:77:in `real_const_get'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/constant_cache.rb:157:in `block in dfs_module'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/constant_cache.rb:145:in `each'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/constant_cache.rb:145:in `dfs_module'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/constant_cache.rb:70:in `initialize'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/gem-generator-tracepoint/tracer.rb:40:in `new'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/gem-generator-tracepoint/tracer.rb:40:in `new'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/hidden-definition-finder.rb:68:in `constant_cache'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/hidden-definition-finder.rb:75:in `all_modules_and_aliases'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/hidden-definition-finder.rb:44:in `main'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/hidden-definition-finder.rb:38:in `main'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/bin/srb-rbi:232:in `block in make_step'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/bin/srb-rbi:121:in `init'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/bin/srb-rbi:196:in `main'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/bin/srb-rbi:237:in `<main>'
/home/mihyaeru/.anyenv/envs/rbenv/versions/3.1.1/lib/ruby/3.1.0/set/sorted_set.rb:2:in `require': cannot load such file -- sorted_set (LoadError)
        from /home/mihyaeru/.anyenv/envs/rbenv/versions/3.1.1/lib/ruby/3.1.0/set/sorted_set.rb:2:in `<top (required)>'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/real_stdlib.rb:77:in `require'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/real_stdlib.rb:77:in `const_get'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/real_stdlib.rb:77:in `call'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/real_stdlib.rb:77:in `real_const_get'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/constant_cache.rb:157:in `block in dfs_module'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/constant_cache.rb:145:in `each'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/constant_cache.rb:145:in `dfs_module'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/constant_cache.rb:70:in `initialize'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/gem-generator-tracepoint/tracer.rb:40:in `new'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/gem-generator-tracepoint/tracer.rb:40:in `new'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/hidden-definition-finder.rb:68:in `constant_cache'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/hidden-definition-finder.rb:75:in `all_modules_and_aliases'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/hidden-definition-finder.rb:44:in `main'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/hidden-definition-finder.rb:38:in `main'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/bin/srb-rbi:232:in `block in make_step'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/bin/srb-rbi:121:in `init'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/bin/srb-rbi:196:in `main'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/bin/srb-rbi:237:in `<main>'
```

In Ruby 3, `SortedSet` is now excluded from `set`.
It needs to explicitly add `sorted_set` to the dependency.
Add the following to `Gemfile` and move on next error.

```
gem 'sorted_set'
```

##### Error 3

```
$ bundle exec srb init
...
Would you like to continue? [Y/n] y
Generating: sorbet/config
Generating: sorbet/rbi/sorbet-typed/
Generating: sorbet/rbi/gems/
LoadError: cannot load such file -- sorbet
LoadError: cannot load such file -- sorbet-static
Generating: sorbet/rbi/hidden-definitions/
Requiring all of your code
Naming all Modules
Naming Bundler::ThorGot LoadError when trying to get nested name Bundler::Thor::RakeCompat
Naming PathnameGot NameError when trying to get nested name Pathname::FileUtils_
Generating /tmp/d20220219-31099-6vnju8/reflection.rbi with 863 modules and 37 aliases
/home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/hidden-definition-finder.rb:401:in `capture_stderr': uninitialized constant Sorbet::Private::HiddenMethodFinder::StringIO (NameError)

    $stderr = StringIO.new
              ^^^^^^^^
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/hidden-definition-finder.rb:89:in `gen_source_rbi'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/hidden-definition-finder.rb:45:in `main'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/hidden-definition-finder.rb:38:in `main'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/bin/srb-rbi:232:in `block in make_step'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/bin/srb-rbi:121:in `init'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/bin/srb-rbi:196:in `main'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/bin/srb-rbi:237:in `<main>'
```

Add the following to `vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/hidden-definition-finder.rb` and move on next error.

```
require 'stringio'
```

##### Error 4

```
$ bundle exec srb init
...
Would you like to continue? [Y/n] y
Generating: sorbet/config
Reusing existing config file: sorbet/config
Generating: sorbet/rbi/sorbet-typed/
Generating: sorbet/rbi/gems/
LoadError: cannot load such file -- sorbet
LoadError: cannot load such file -- sorbet-static
Generating: sorbet/rbi/hidden-definitions/
Requiring all of your code
Naming all Modules
Naming Bundler::ThorGot LoadError when trying to get nested name Bundler::Thor::RakeCompat
Naming PathnameGot NameError when trying to get nested name Pathname::FileUtils_
Generating /tmp/d20220219-2706-9me751/reflection.rbi with 866 modules and 37 aliases
Printing your code's symbol table into /tmp/d20220219-2706-9me751/from-source.json
Printing /tmp/d20220219-2706-9me751/reflection.rbi's symbol table into /tmp/d20220219-2706-9me751/reflection.json
/home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/hidden-definition-finder.rb:153:in `write_constants': /tmp/d20220219-2706-9me751/reflection.rbi had unexpected errors. Check this file for a clue: /tmp/d20220219-2706-9me751/reflection.json.err (RuntimeError)
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/hidden-definition-finder.rb:48:in `main'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/lib/hidden-definition-finder.rb:39:in `main'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/bin/srb-rbi:232:in `block in make_step'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/bin/srb-rbi:121:in `init'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/bin/srb-rbi:196:in `main'
        from /home/mihyaeru/ghq/github.com/mihyaeru21/sorbet-init-error/vendor/ruby/3.1.0/gems/sorbet-0.5.9663/bin/srb-rbi:237:in `<main>'

$ cat /tmp/d20220219-2706-9me751/reflection.json.err
/tmp/d20220219-2706-9me751/reflection.rbi:14763: Parent of class `Random` redefined from `Object` to `Random::Base` https://srb.help/5012
       14763 |class Random < Random::Base
                             ^^^^^^^^^^^^
```

This is reported by https://github.com/sorbet/sorbet/issues/3851#issuecomment-751403792.
In Ruby 3, `Random` inherits from `Random::Base`.
In Ruby 2.7, it inherits from `Object`.
[Sorbet's rbi](https://github.com/sorbet/sorbet/blob/d2c97f0e57238ddee8aab61fd176d99651e7b76d/rbi/core/random.rbi#L31) is Ruby 2.7 compliant and inherits from `Object`.

I couldn't figure out how to fix this.

#### Expected behavior
No error has raised and `srb init` has completed.

### PR DESCRIPTION
**Pull request by elliottt**
Resolve the remaining issues outlined in #5332, and add ruby-3.0.3 to the set that we test `srb init` with.

Fixes #5332

### Motivation

Fixing bugs related to ruby-3.

### Test plan

See included automated tests.

### CODE REVIEW

File: gems/sorbet/test/hidden-method-finder/BUILD [hunk 1]
```diff
@@ -35,10 +36,19 @@ hidden_methods_tests(
...
+    tests = [
+        "simple",
+        "thorough",
+    ],
+)
+
 test_suite(
     name = "hidden-method-finder",
     tests = [
         ":hidden-methods-sorbet_ruby_2_6",
```
Comment (jez) [inline gems/sorbet/test/hidden-method-finder/BUILD:50]: Ruby 2.6 goes EOL in 2 days. Do you want to just delete the tests now to save time?

https://www.ruby-lang.org/en/downloads/branches/

**Review by jez — APPROVED** [review]
Looks great, thanks!

ooc, did you happen to try locally with 3.1 as well? i'm just curious whether we need further fixes to get it working with 3.1

**Comment by elliottt** [comment]
> Looks great, thanks!
> 
> ooc, did you happen to try locally with 3.1 as well? i'm just curious whether we need further fixes to get it working with 3.1

I did, and it worked with 3.1 as well.

File: gems/sorbet/test/hidden-method-finder/BUILD (hunk 1 above)
Comment (elliottt) [reply gems/sorbet/test/hidden-method-finder/BUILD:50]: We still depend on it for rubyfmt, so I'll just remove the testing that depends on it for now.

**Pull request merged**